src/library/serato/seratofeature.cpp

```diff
@@ -0,0 +1,1043 @@
+
+        // Read field data
+        QByteArray data = buffer.read(fieldSize);
+        if (static_cast<quint32>(data.length()) != fieldSize) {
+            qWarning() << "Failed to read "
+                       << fieldSize
+                       << " bytes for "
+                       << fieldName
+                       << " field.";
+            return false;
+        }
+
+        // Parse field data
+        switch (static_cast<FieldId>(fieldId)) {
+        case FieldId::FileType:
+            track.filetype = parseText(data, fieldSize);
+            break;
+        case FieldId::FilePath:
+            track.location = parseText(data, fieldSize);
+            break;
+        case FieldId::SongTitle:
+            track.title = parseText(data, fieldSize);
+            break;
+        case FieldId::Artist:
+            track.artist = parseText(data, fieldSize);
+            break;
+        case FieldId::Album:
+            track.album = parseText(data, fieldSize);
+            break;
+        case FieldId::Genre:
+            track.genre = parseText(data, fieldSize);
+            break;
+        case FieldId::Length: {
+            bool ok;
+            int duration = parseText(data, fieldSize).toInt(&ok);
+            if (ok) {
+                track.duration = duration;
+            }
+            break;
+        }
+        case FieldId::Bitrate:
+            track.bitrate = parseText(data, fieldSize);
+            break;
+        case FieldId::SampleRate:
+            track.samplerate = parseText(data, fieldSize);
+            break;
+        case FieldId::Bpm: {
+            bool ok;
+            double bpm = parseText(data, fieldSize).toDouble(&ok);
+            if (ok) {
+                track.bpm = bpm;
+            }
+            break;
+        }
+        case FieldId::Comment:
+            track.comment = parseText(data, fieldSize);
+            break;
+        case FieldId::Grouping:
+            track.grouping = parseText(data, fieldSize);
+            break;
+        case FieldId::Label:
+            track.label = parseText(data, fieldSize);
+            break;
+        case FieldId::Year: {
+            bool ok;
+            int year = parseText(data, fieldSize).toInt(&ok);
+            if (ok) {
+                track.year = year;
+            }
+            break;
+        }
+        case FieldId::Key:
+            track.key = parseText(data, fieldSize);
+            break;
+        case FieldId::BeatgridLocked:
+            track.beatgridlocked = parseBoolean(data);
+            break;
+        case FieldId::Missing:
+            track.missing = parseBoolean(data);
+            break;
+        case FieldId::FileTime:
+            track.filetime = parseUInt32(data);
+            break;
+        case FieldId::DateAdded:
+            track.datetimeadded = parseUInt32(data);
+            break;
+        case FieldId::DateAddedText:
+            // Ignore this field, but do not print a debug message
```

Comment:
Why?

Reply:
It's the same as the regular DateAdded field, but this time the timestamp is a utf16 string instead of an int. Since we already parse the int version, it doesn't make sense to parse the string field.